packages/supermassive/src/hooks/types.ts

```diff
@@ -71,11 +71,13 @@ export interface AfterBuildResponseHook<ResolveContext = unknown> {
 }
 
 export interface BeforeOperationExecuteHook<ResolveContext = unknown> {
-  (args: BaseExecuteOperationHookArgs<ResolveContext>): void;
+  (args: BaseExecuteOperationHookArgs<ResolveContext>): void | Promise<void>;
```

Comment:
What about `BeforeFieldResolveHook`? Should we change return type for it as well?